sdk/core/azure-core/src/main/java/com/azure/core/http/HttpPipelineBuilder.java

```diff
@@ -73,7 +78,22 @@ public HttpPipeline build() {
             client = HttpClient.createDefault();
         }
 
-        return new HttpPipeline(client, policies);
+        configureTracing(policies);
+
+        return new HttpPipeline(client, policies, tracer);
+    }
+
+    private void configureTracing(List<HttpPipelinePolicy> policies) {
+        if (tracer == null) {
+            TracingOptions tracingOptions = clientOptions == null ? null : clientOptions.getTracingOptions();
+            tracer = TracerProvider.getDefaultProvider().createTracer("azure-core", null, null, tracingOptions);
+        }
+
+        for (HttpPipelinePolicy policy : policies) {
+            if (policy instanceof InstrumentationPolicy) {
+                ((InstrumentationPolicy) policy).initialize(tracer);
```

Comment:
If the pipeline doesn't have an InstrumentationPolicy will the Tracer ever be used?

Reply:
yes, by `RestProxy` - this is why I added `getTracer` method.